typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
+      existingIsmAddress,
+      this.multiProvider.getSigner(destination),
+    );
+
+    // Enroll domains
+    for (const originDomain of delta.domainsToEnroll) {
+      const origin = this.multiProvider.getChainName(originDomain);
+      logger.debug(
+        `Reconfiguring preexisting routing ISM for origin ${origin}...`,
+      );
+      const ism = await EvmIsmModule.deploy({
+        destination,
+        config: config.domains[origin],
+        origin,
+        mailbox,
+        multiProvider: this.multiProvider,
+        factories: this.args.addresses,
+        deployer: this.deployer,
+      });
+      const tx = await routingIsm.set(originDomain, ism.address, overrides);
+      await this.multiProvider.handleTx(destination, tx);
+    }
+
+    // Unenroll domains
+    for (const originDomain of delta.domainsToUnenroll) {
+      logger.debug(
+        `Unenrolling originDomain ${originDomain} from preexisting routing ISM at ${existingIsmAddress}...`,
+      );
+      const tx = await routingIsm.remove(originDomain, overrides);
+      await this.multiProvider.handleTx(destination, tx);
+    }
+
+    // Transfer ownership if needed
+    if (delta.owner) {
+      logger.debug(`Transferring ownership of routing ISM...`);
+      const tx = await routingIsm.transferOwnership(delta.owner, overrides);
+      await this.multiProvider.handleTx(destination, tx);
+    }
+
+    return routingIsm;
+  }
+
+  protected async initializeFallbackRoutingIsm(params: {
+    destination: ChainName;
+    owner: Address;
+    safeConfigDomains: Domain[];
+    submoduleAddresses: Address[];
+    existingIsmAddress: Address;
+    signer: Signer;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const {
+      destination,
+      owner,
+      safeConfigDomains,
+      submoduleAddresses,
+      existingIsmAddress,
+      signer,
+      logger,
+    } = params;
+    const routingIsm = DefaultFallbackRoutingIsm__factory.connect(
+      existingIsmAddress,
+      signer,
+    );
+
+    logger.debug('Initializing fallback routing ISM ...');
+    await this.multiProvider.handleTx(
+      destination,
+      routingIsm['initialize(address,uint32[],address[])'](
+        owner,
+        safeConfigDomains,
+        submoduleAddresses,
+        this.multiProvider.getTransactionOverrides(destination),
+      ),
+    );
+
+    return routingIsm;
+  }
+
+  protected async deployRoutingIsmSubmodules(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    mailbox?: Address;
+  }): Promise<Address[]> {
+    const { destination, config, mailbox } = params;
+    const isms: ChainMap<Address> = {};
+
+    for (const origin of Object.keys(config.domains)) {
+      const ism = await EvmIsmModule.deploy({
+        destination,
+        config: config.domains[origin],
+        origin,
+        mailbox,
+        multiProvider: this.multiProvider,
+        factories: this.args.addresses,
+        deployer: this.deployer,
```

Comment:
seems like we have to pass a lot of context through
should we make the deploy fn non-static?